sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/CodeSnippetsAysnc.java

```diff
@@ -0,0 +1,146 @@
+package com.azure.data.tables;
+
+
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+
+
+public class CodeSnippetsAysnc {
+
+
+    public void AsyncSnippets() {
+
+        //build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceAsyncClientBuilder()
+            .connectionString("connectionString")
+            .build();
+
+        //add table
+        Mono<TableAsyncClient> createTableMono = tableServiceAsyncClient.createTable("OfficeSupplies");
+
+        createTableMono.subscribe(Void -> {
+            System.out.println("Table creation successful.");
+        }, error -> {
+            System.out.println("There was an error creating the table. Error: " + error);
+        });
+
+
+
+        //delete a table
+        Mono<Void> deleteTableMono = tableServiceAsyncClient.deleteTable("OfficeSupplies");
+
+        deleteTableMono.subscribe(Void -> {
+            System.out.println("Table deletion successful");
+        }, error -> {
+            System.out.println("There was an error deleting the table. Error: " + error);
+        });
+
+
+        //query tables
+        String selectString = "$select= TableName eq 'OfficeSupplies'";
+        Flux<AzureTable> queryTableFlux = tableServiceAsyncClient.queryTables(selectString);
+        queryTableFlux.subscribe(azureTable -> {
+            System.out.println(azureTable.getName());
+        }, error -> {
+            System.out.println("There was an error querying the service. Error: " + error);
+        });
+    }
+
+    public void InsertEntity(){
+
+        //build service client
+        TableServiceAsyncClient tableServiceAsyncClient = new TableServiceAsyncClientBuilder()
+            .connectionString("connectionString")
+            .build();
+
+        TableAsyncClient tableAsyncClient = tableServiceAsyncClient.getClient("OfficeSupplies");
+
+
+        String tableName = "OfficeSupplies";
+        String row = "crayolaMarkers";
+        String partitionKey = "markers";
+        HashMap<String, Object> tableEntityProperties = new HashMap<>();
+        Mono<TableEntity> insertEntityMono = tableAsyncClient.insertEntity(tableName, row, partitionKey,
```

Comment:
You don't need to create a new variable (ie. insertEntityMono) for these Mono operations. It makes it harder to read.  Same for other instances where you create the Mono variable to link it to a subscribe.
You can use:
```java
tableAsyncClient.insertEntity(tableName, row, partitionKey, tableEntityProperties).subscribe(tableEntity -> {
       // Stuff
   }, error -> {
      // Stuff
   });
```